syntax = "proto3";
import "google/protobuf/empty.proto";
option go_package = ".;proto";

service User  {
  rpc GetUserByPhone(PhoneRequest) returns (UserInfoResponse);//通过手机号查询用户
  rpc GetUserList(PageInfo) returns (UserListResponse);//用户列表
  rpc GetUserByID(UserIDRequest) returns (UserInfoResponse);//通过id查询用户
  rpc CreateUser(CreateUserInfo) returns (UserInfoResponse);//创建用户
  rpc UpdateUser(UpdateUserInfo) returns (google.protobuf.Empty);//更新用户信息
  rpc CheckPassword(PasswordCheckInfo) returns (CheckResponse);//检查用户密码
}
message PasswordCheckInfo {
  string password = 1;
  string  encryptedPassword = 2;
}
message PageInfo{
  uint32 pn = 1;
  uint32 pSize = 2;

}
message CheckResponse {
  bool success = 1;
}
message PhoneRequest {
  string phone = 1;
}
message UserIDRequest {
  int64 id = 1;
}
message CreateUserInfo {
  string nickName = 1;
  string password = 2;
  string phone = 3;
}
message UpdateUserInfo {
  int64 id = 1;
  string nickName = 2;
  string gender = 3;
  uint64 birthDay = 4;
}
//返回信息

message UserInfoResponse {
  int64 id = 1;
  string nickName = 2;
  string password = 3;
  string phone = 4;
  uint64 birthDay = 5;
  string gender = 6;
  int32 role = 7;
}


message UserListResponse {
  int32 total = 1;
  repeated UserInfoResponse data = 2;
}